### PR TITLE
add typ values

### LINKED ISSUE #30: typ values for Client Attestation JWT and Client Attestation PoP JWT
The current draft does not require the Client Attestation JWT and Client Attestation PoP JWT to have the `typ` JWS header parameter. Do you have any plans to define specific `typ` values for these JWTs?

### LINKED ISSUE #72: Add required typ values for the client-attestation and client-attestation-pop in line with JWT BCP

### PR DESCRIPTION
**Pull request by paulbastian**
Closes #30 
Closes #72

### CODE REVIEW

**Comment by c2bo** [comment]
This includes #90 which makes it pretty hard to see the changes to main :/ I guess let's wait until 90 is merged?

File: draft-ietf-oauth-attestation-based-client-auth.md [hunk 1]
```diff
@@ -118,83 +120,40 @@ Client Instance:
...
+* `exp`: REQUIRED. The `exp` (expiration time) claim MUST specify the time at which the Client Attestation is considered expired by its issuer. The authorization server MUST reject any JWT with an expiration time that has passed, subject to allowable clock skew between systems.
+* `cnf`: REQUIRED. The `cnf` (confirmation) claim MUST specify a key conforming to {{RFC7800}} that is used by the Client Instance to generate the Client Attestation PoP JWT for client authentication with an authorization server. The key MUST be expressed using the "jwk" representation.
+* `iat`: OPTIONAL. The `iat` (issued at) claim MUST specify the time at which the Client Attestation was issued.
+* `nbf`: OPTIONAL. The `nbf` (not before) claim MUST specify the time before which the Client Attestation MUST NOT be accepted for processing.
 
-5. The JWT MAY contain an "nbf" (not before) claim that identifies the time before which the token MUST NOT be accepted for processing.
+The following additional rules apply:
 
-6. The JWT MAY contain an "iat" (issued at) claim that identifies the time at which the JWT was issued.
+1. The JWT MAY contain other claims.
```
Comment (c2bo) [inline draft-ietf-oauth-attestation-based-client-auth.md:146]: Should we explain what happens if claims are not understood by a RP?
We have 2 options and should probably clearly state how to deal with such a situation in the spec:
- ignore not understood claims
- fail validation of the client attestation

File: draft-ietf-oauth-attestation-based-client-auth.md [hunk 2]
```diff
@@ -218,34 +177,37 @@ The following example is the decoded header and payload of a JWT meeting the pro
...
+* `nonce`: OPTIONAL. The `nonce` (nonce) claim MUST specify a String value that is provided by the authorization server to associate the Client Attestation PoP JWT with a particular transaction and prevent replay attacks.
+* `iat`: OPTIONAL. The `iat` (issued at) claim MUST specify the time at which the Client Attestation PoP was issued. Note that the authorization server may reject JWTs with an "iat" claim value that is unreasonably far in the past.
+* `nbf`: OPTIONAL. The `nbf` (not before) claim MUST specify the time before which the Client Attestation PoP MUST NOT be accepted for processing.
 
-7. The JWT MAY contain an "iat" (issued at) claim that identifies the time at which the JWT was issued.  Note that the authorization server may reject JWTs with an "iat" claim value that is unreasonably far in the past.
+The following additional rules apply:
 
-8. The JWT MAY contain other claims.
+1. The JWT MAY contain other claims.
```
Comment (c2bo) [inline draft-ietf-oauth-attestation-based-client-auth.md:198]: Same as above: How to deal with claims that are not understood by a RP?

File: draft-ietf-oauth-attestation-based-client-auth.md (hunk 1 above)
Comment (c2bo) [reply draft-ietf-oauth-attestation-based-client-auth.md:157]:
```suggestion
1. The JWT MAY contain other claims. All claims that are not understood by implementations MUST be ignored.
```

File: draft-ietf-oauth-attestation-based-client-auth.md (hunk 2 above)
Comment (c2bo) [reply draft-ietf-oauth-attestation-based-client-auth.md:209]:
```suggestion
1. The JWT MAY contain other claims. All claims that are not understood by implementations MUST be ignored.
```

**Review by c2bo — APPROVED** [review]

**Review by tplooker — APPROVED** [review]

**Pull request merged**